packages/realm_dart/lib/src/map.dart

```diff
@@ -273,12 +287,23 @@ extension RealmMapInternal<T extends Object?> on RealmMap<T> {
 class MapNotificationsController<T extends Object?> extends NotificationsController {
   final ManagedRealmMap<T> map;
   late final StreamController<RealmMapChanges<T>> streamController;
+  List<String>? keyPaths;
+
+  MapNotificationsController(this.map, [List<String>? keyPaths]) {
+    if (keyPaths != null) {
+      this.keyPaths = keyPaths;
+
+      if (keyPaths.any((element) => element.isEmpty || element.trim().isEmpty)) {
+        throw RealmException("A key path cannot be empty or consisting only of white spaces");
+      }
```

Comment:
We can definitely move it to `verifyKeyPath`, and yes I can improve the wording. 
But no, core won't complain, and I thought it would be better to throw an exception for this.